pkg/capi/aws.go

```diff
@@ -341,3 +338,84 @@ func newAWSMachineTemplate(name string, mapiProviderSpec *mapiv1.AWSMachineProvi
 
 	return awsMachineTemplate
 }
+
+// createCAPIMachineSetWithRetry creates a CAPI MachineSet with retry logic for capacity constraints.
+// It tries different instance types when encountering insufficient capacity errors.
+func createCAPIMachineSetWithRetry(ctx context.Context, cl client.Client, machineSetName string, clusterName string, platform configv1.PlatformType, mapiDefaultProviderSpec *mapiv1.AWSMachineProviderConfig, templateConfigurator func(*awsv1.AWSMachineTemplate, string)) (*clusterv1.MachineSet, *awsv1.AWSMachineTemplate) {
```

Comment:
Should we call it createAWSCAPIMachineSetWithRetry considering it is AWS specific?